From a215c3b1b2bddceb47b8a7ebc013a77d43b7fcde Mon Sep 17 00:00:00 2001
From: Balaji Eswaramurthy <beswaramurthy@maxlinear.com>
Date: Thu, 22 Aug 2024 17:40:04 +0800
Subject: [PATCH] WLANRTSYS-84214 Send reassoc notification to FW only after
 UMI_ADD_STA_MLD

Issue:
Reassociation request is received from station after cleanup is done in
the firmware. When processing this reassoc notification firmware is
asserted.

Fix:
Send reassociation notification to firmware only after UMI_ADD_STA_MLD
is done.

FIXUP to WLANRTSYS-62845 MLO changes: MLO Auth Assoc, WPA3 and eapol
changes
---
 src/ap/mxl_ieee802_11.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/ap/mxl_ieee802_11.c b/src/ap/mxl_ieee802_11.c
index a73dce32a..01dea25d3 100644
--- a/src/ap/mxl_ieee802_11.c
+++ b/src/ap/mxl_ieee802_11.c
@@ -384,7 +384,8 @@ void mxl_mlo_reassoc_notify(struct hostapd_data *hapd, struct sta_info *sta)
 {
 	if (mxl_is_mlo_enabled(hapd->conf) &&
 			sta && mxl_mlo_is_ml_sta(sta) &&
-			(sta->flags & WLAN_STA_ASSOC)) {
+			(sta->flags & WLAN_STA_ASSOC) &&
+			(sta->mxl_sta_info.ext_flags & WLAN_STA_MLD_STA)) {
 		struct mxl_ml_sta_reassoc_notify reassoc_notify;
 
 		reassoc_notify.aid = sta->aid;
-- 
2.43.0

